pkg/pillar/cmd/volumemgr/sizemgmt.go

```diff
@@ -131,3 +134,25 @@ func persistUsageStat(_ *volumemgrContext) (*types.UsageStat, error) {
 	}
 	return usageStat, nil
 }
+
+// dirUsage calculates usage of directory
+// it checks if provided directory is zfs mountpoint and take usage from zfs in that case
+func dirUsage(_ *volumemgrContext, dir string) (uint64, error) {
+	if vault.ReadPersistType() != types.PersistZFS || !strings.HasPrefix(dir, types.PersistDir) {
+		return diskmetrics.SizeFromDir(log, dir)
+	}
+	mi, err := mount.Lookup(dir)
+	if err != nil {
+		log.Errorf("dirUsage: Lookup returns error (%s), fallback to SizeFromDir", err)
```

Comment:
Don't you see this errors since we are reporting e.g., /persist/tmp and /persist/certs which are not separate datasets/mounts?

Reply:
It is loop over all separate parts of the path internally and return longest one, so, no.

Reply:
Correct, it [iterates](https://github.com/moby/sys/blob/main/mountinfo/mountinfo_linux.go#L111-L122) over all mountpoints and apply [filter](https://github.com/moby/sys/blob/main/mountinfo/mountinfo_filters.go#L48) which is check if mountpoint is a prefix for provided dir. And finally returns the longest found mountpoint. So for your sample if /persist/certs is not mountpoint, but /persist is mountpoint, it will return /persist. If both are not mountpoints, it will return /.